src/validation/statement.rs

```diff
@@ -309,9 +326,16 @@ fn validate_cast_literal<T: AnnotationMap>(
     );
 
     if !literal.is_cast_prefix_eligible() {
-        validator.push_diagnostic(Diagnostic::literal_expected(location.clone()))
+        validator.push_diagnostic(
+            Diagnostic::error(format!(
+                "Cannot cast into {}, only elementary types are allowed",
+                literal.get_literal_value()
```

Comment:
Not sure if this makes things better but maybe `validator.context.slice(&statement.get_location())`; atleast we won't get a debug-string for literal arrays